Cargo.lock

```diff
@@ -1522,7 +1522,7 @@ dependencies = [
  "libc",
  "log",
  "rustversion",
- "windows",
+ "windows 0.54.0",
```

Comment:
might be worth digging what still uses 0.54 here and whether we could avoid the duplication, but doesn't have to be now

Reply:
This is actually easy to answer with `cargo tree`:

```
$ cargo tree -i windows@0.54.0 -i windows@0.58.0 --all-features --all-targets

windows v0.54.0
└── generator v0.8.1
    └── loom v0.7.2
        └── tracy-client v0.17.1
            └── wgpu-benchmark v22.0.0 (E:\wgpu\benches)

windows v0.58.0
├── gpu-allocator v0.27.0
│   └── wgpu-hal v22.0.0 (E:\wgpu\wgpu-hal)
│       ├── wgpu v22.0.0 (E:\wgpu\wgpu)
│       │   ├── wgpu-benchmark v22.0.0 (E:\wgpu\benches)
│       │   ├── wgpu-examples v22.0.0 (E:\wgpu\examples)
│       │   ├── wgpu-info v22.0.0 (E:\wgpu\wgpu-info)
│       │   └── wgpu-test v22.0.0 (E:\wgpu\tests)
│       │       [dev-dependencies]
│       │       └── wgpu-examples v22.0.0 (E:\wgpu\examples)
│       └── wgpu-core v22.0.0 (E:\wgpu\wgpu-core)
│           ├── player v22.0.0 (E:\wgpu\player)
│           └── wgpu v22.0.0 (E:\wgpu\wgpu) (*)
└── wgpu-hal v22.0.0 (E:\wgpu\wgpu-hal) (*)
```

Reply:
`windows` 0.52 is what's currently in use in `mozilla-central`, and it has explicit use by _many_ crates. I'm not sure how easy this is to upgrade in `mozilla-central`, but I suspect it's non-trivial, _unless_ `windows` 0.52 is source-compatible _enough_ with 0.58.0.

CC @glandium and @jimblandy, who I'm sure would be interested in the pressure we're feeling on this front.

Reply:
> might be worth digging what still uses 0.54

No need to dig, it's pretty easy to see from the `Cargo.lock` context (or `cargo tree` otherwise) that this is [`generator`](https://crates.io/crates/generator).

`cargo tree` however points out that it's a seemingly-irrelevant dependency, exclusively involved in the `wgpu-benchmark` tree and not clobbering `wgpu`/`wgpu-hal`?

As for `0.52`: that really is quite a ways back and very unlikely to be version-range-compatible, especially for `Direct3D12`. It'll definitely work for our purposes, though. Let me know if you want me to downgrade in all PRs.

Reply:
`windows-sys` 0.52 is what's used by plenty of crates. `windows` 0.52 is only used by gpu-allocator. Note the versions are *NOT* correlated. 0.52 is the current version for `windows-sys` and `windows` 0.58 depends on it. Anyways, there is some manual work involved in upgrading `windows` in mozilla-central, but nothing prevents it from happening.

Reply:
@glandium note that `windows` doesn't depend on `windows-sys`. It depends on `windows-targets`. Which `windows-sys` also depends on.

It is (unfortunately) not a classical `-sys` setup that helps deduplicating low-level definitions by splitting them out of the `windows` crate into `windows-sys` (i.e. having vtable function definitions there would be immensely helpful).

Reply:
Sure, but that doesn't change the version requirements for mozilla-central.

Reply:
> 0.52 is only used by gpu-allocator.

`gpu-allocator` is a crate that I maintain and recently released a new version for with support all the way up to `windows 0.58`.

The `mozilla-central` repo indeed [uses `windows 0.52`](https://searchfox.org/mozilla-central/rev/de46cd99a7d1634058b511a3f546a970763e5048/Cargo.lock#6915-6916) [for `gpu-allocator`](https://searchfox.org/mozilla-central/rev/de46cd99a7d1634058b511a3f546a970763e5048/Cargo.lock#2562) but it's [only used by `wgpu-hal`](https://searchfox.org/mozilla-central/rev/de46cd99a7d1634058b511a3f546a970763e5048/Cargo.lock#6819) whose version I am bumping **in this PR**, so this should be a **non-issue**.

It seems you [have a semver-compatible patch for `windows-targets`](https://searchfox.org/mozilla-central/rev/de46cd99a7d1634058b511a3f546a970763e5048/Cargo.toml#110-112) anyway.

Reply:
Heh, nice!